scipy/stats/morestats.py

```diff
@@ -3147,16 +3147,32 @@ def median_test(*args, **kwds):
     return stat, p, grand_median, table
 
 
-def _circfuncs_common(samples, high, low):
+def _circfuncs_common(samples, high, low, **kwds):
+    # Ensure samples are array-like
     samples = np.asarray(samples)
+
+    # Read and apply the NaN policy, defaulting to propagate (previous behavior)
+    nan_policy = kwds.pop('nan_policy', 'propagate')
```

Comment:
There's no reason here as far as I can see to not just add a `nan_policy` parameter. It doesn't even have to be optional, can be a required positional argument since all public function have it as a keyword.

This may have come from some other place in SciPy perhaps where it was necessary?